src/js_realm.hpp

```diff
@@ -874,61 +888,50 @@ void RealmClass<T>::async_open_realm(ContextType ctx, ObjectType this_object, Ar
         Function<T>::callback(protected_ctx, protected_callback, protected_this, 1, callback_arguments);
     }
 
-    // First download the Realm with no schema set to avoid spurious writes that
-    // the server may either reject (for read-only Realms) or just waste time
-    // merging.
-    std::shared_ptr<Realm> realm;
-    try {
-        auto download_config = config;
-        download_config.schema = util::none;
-        download_config.cache = false;
-        realm = realm::Realm::get_shared_realm(std::move(download_config));
-    }
-    catch (const RealmFileException& ex) {
-        handleRealmFileException(ctx, config, ex);
-    }
-
-    auto session = user->session_for_on_disk_path(realm->config().path);
-    EventLoopDispatcher<WaitHandler> wait_handler([=, config=std::move(config),
-                                                   defaults=std::move(defaults),
-                                                   constructors=std::move(constructors)](std::error_code error_code) mutable {
+    std::shared_ptr<AsyncOpenTask>* ptr = new std::shared_ptr<AsyncOpenTask>(Realm::get_synchronized_realm(config));
```

Comment:
Why do you need to create a pointer to a shared pointer?

Reply:
I don't remember the specifics, but I ran into issues when not doing this. Also, it seems that SharedRealm uses a similar pattern.

But I did refactor a few things after this, so I can try to remove it again.

Reply:
It's a common silly-looking pattern in the JS stuff. the JS wrapper object stores a `unique_ptr<T>`, so we end up with a `unique_ptr<shared_ptr<Realm>>` when we actually want shared ownership.